spacy/pipeline/spancat.py

```diff
@@ -408,16 +408,17 @@ def _make_span_group(
         spans = SpanGroup(doc, name=self.key)
         max_positive = self.cfg["max_positive"]
         threshold = self.cfg["threshold"]
+
+        keeps = scores >= threshold
+        ranked = (scores * -1).argsort()
+        keeps[ranked[:, max_positive:]] = False
+        spans.attrs["scores"] = scores[keeps].flatten()
```

Comment:
While we're here, let's do:

```
indices = to_numpy(indices)
keeps = to_numpy(keeps)
```

Before the loop. We can then remove the `int()` calls around the array access. It's worth noting that if we ever have to call `int()` on individual items to copy back to CPU, we're probably doing something wrong --- we shouldn't be syncing from GPU to CPU within a loop, or it'll be really slow.